vortex-api/src/main/java/mil/army/usace/hec/vortex/util/IndexMap.java

```diff
@@ -0,0 +1,24 @@
+package mil.army.usace.hec.vortex.util;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public final class IndexMap<K> {
+    private final Map<K, Integer> internalMap;
+
+    private IndexMap(Map<K, Integer> internalMap) {
+        this.internalMap = internalMap;
+    }
+
+    public static <K> IndexMap<K> of(List<K> sortedList) {
```

Comment:
Could this be relocated to the package of it's usage so that it does not have to be public API?

Reply:
Migrated to `io` package and made package-private